articles/notification-hubs/notification-hubs-android-push-notification-google-fcm-get-started.md

```diff
@@ -175,134 +153,79 @@ Your notification hub is now configured to work with Firebase Cloud Messaging, a
-                        // If you want to use tags...
-                        // Refer to : https://azure.microsoft.com/documentation/articles/notification-hubs-routing-tag-expressions/
-                        // regID = hub.register(token, "tag1,tag2").getRegistrationId();
-   
-                        resultString = "New NH Registration Successfully - RegId : " + regID;
-                        Log.d(TAG, resultString);
-   
-                        sharedPreferences.edit().putString("registrationID", regID ).apply();
-                        sharedPreferences.edit().putString("FCMtoken", FCM_token ).apply();
-                    }
-   
-                    // Check if the token may have been compromised and needs refreshing.
-                    else if ((storedToken=sharedPreferences.getString("FCMtoken", "")) != FCM_token) {
-   
-                        NotificationHub hub = new NotificationHub(NotificationSettings.HubName,
-                                NotificationSettings.HubListenConnectionString, this);
-                        Log.d(TAG, "NH Registration refreshing with token : " + FCM_token);
-                        regID = hub.register(FCM_token).getRegistrationId();
-   
-                        // If you want to use tags...
-                        // Refer to : https://azure.microsoft.com/documentation/articles/notification-hubs-routing-tag-expressions/
-                        // regID = hub.register(token, "tag1,tag2").getRegistrationId();
-   
-                        resultString = "New NH Registration Successfully - RegId : " + regID;
-                        Log.d(TAG, resultString);
-   
-                        sharedPreferences.edit().putString("registrationID", regID ).apply();
-                        sharedPreferences.edit().putString("FCMtoken", FCM_token ).apply();
-                    }
-   
-                    else {
-                        resultString = "Previously Registered Successfully - RegId : " + regID;
-                    }
+
+                    NotificationHub hub = new NotificationHub(NotificationSettings.HubName,
+                            NotificationSettings.HubListenConnectionString, this);
+
+                    Log.d(TAG, "NH Registration refreshing with token : " + FCM_token);
+                    String regID = hub.register(FCM_token).getRegistrationId();
+
+                    // If you want to use tags...
+                    // Refer to : https://azure.microsoft.com/en-us/documentation/articles/notification-hubs-routing-tag-expressions/
+                    // regID = hub.register(token, "tag1,tag2").getRegistrationId();
+
+                    resultString = "New NH Registration Successfully - RegId : " + regID;
+                    Log.d(TAG, resultString);
                 } catch (Exception e) {
-                    Log.e(TAG, resultString="Failed to complete registration", e);
+                    Log.e(TAG, resultString = "Failed to complete registration: " + e.toString());
```

Comment:
It appears that the `Log.e` calls usually have the third, `tr` parameter set. Since the exception object instance is available to us here, I suggest we pass it too, for uniformity's sake.
```suggestion
                    Log.e(TAG, resultString = "Failed to complete registration: " + e.getMessage(), e);
```

Reply:
done